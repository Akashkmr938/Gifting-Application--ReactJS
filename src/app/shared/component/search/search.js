/* eslint-disable no-use-before-define */
import React, { useState, useRef, useEffect } from 'react';
import TextField from '@material-ui/core/TextField';
import Autocomplete from '@material-ui/lab/Autocomplete';
import searchStyle from './search.module.scss';
import { httpGet } from './../../utils/http/http';
import { withRouter } from 'react-router-dom';

const SearchBox = (props) => {

  const [searchValue, setSearchValue] = useState('');
  const [searchResult, setSearchResult] = useState([]);

  const searchRef = useRef();

  useEffect(() => {
    if (searchRef.current.value.trim() !== '') {
      const timer = setTimeout(() => {
        if (searchValue === searchRef.current.value) {
          httpGet(`/giftCards?q=${searchValue}&_limit=5`)
            .then(response => setSearchResult(response.data))
        }
      }, 500);
      return () => {
        clearTimeout(timer);
      };
    }
  }, [searchValue, searchRef]);

  const searchClickHandler = (event) => {
    event.preventDefault();
    props.history.push(`/searchresults?search=${searchRef.current.value}`);
  }

  return (
    <form onSubmit={searchClickHandler}>
      <Autocomplete
        id="combo-box-demo"
        options={searchResult}
        getOptionLabel={option => option.name}
        style={{ width: 300 }}
        renderInput={params => (
          <TextField inputRef={searchRef} onChange={event => setSearchValue(event.target.value)} className={searchStyle.searchInput} {...params} label="Search" variant="outlined" fullWidth />
        )}
      />
    </form>
  );
}

export default withRouter(SearchBox);