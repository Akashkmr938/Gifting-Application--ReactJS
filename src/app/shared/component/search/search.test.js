import React from 'react';
import renderer from 'react-test-renderer';
import search from './search';

describe('The Search component', () => {
    it('renders correctly', () => {
        const tree = renderer
            .create(<search />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
})