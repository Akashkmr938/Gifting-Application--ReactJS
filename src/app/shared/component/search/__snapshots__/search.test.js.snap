// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Search component renders correctly 1`] = `<search />`;
